### PR TITLE
[SwiftSyntax] Enable Package CMO if possible

### PR DESCRIPTION
**Pull request by rintaro**
Update for https://github.com/swiftlang/swift-syntax/pull/2736

The compiler crashes when compiling swift files importing package-cmo enabled `.swiftmodule`. Avoid that by not-using `.swiftmodule`.

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-syntax/pull/2736
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-syntax/pull/2736
@swift-ci Please smoke test macOS

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-syntax/pull/2736
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-syntax/pull/2736
@swift-ci Please smoke test

File: cmake/modules/SwiftCompilerCapability.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  )
+  if(NOT result)
+    set("${out_var}" "TRUE" PARENT_SCOPE)
+  else()
+    set("${out_var}" "FALSE" PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Test if the Swift compiler supports -disable-implicit-<module>-module-import.
+macro(swift_supports_implicit_module module_name out_var)
```
Comment (edymtt) [inline cmake/modules/SwiftCompilerCapability.cmake:18]: (Optional) There is a function with the same name in `cmake/modules/SwiftImplicitImport.cmake` -- we may consider folding that into this new implementation.

Comment (rintaro) [reply cmake/modules/SwiftCompilerCapability.cmake:18]: This PR removes `cmake/modules/SwiftImplicitImport.cmake` and replaces `swift_supports_implicit_module` function with this new implementation.

**Review by bnbarham — APPROVED** [review]

**Pull request merged**